app/models/message.rb

```diff
@@ -4,7 +4,7 @@ class Message < ApplicationRecord
   has_many :matched_messages
   has_many :subscription_contents
 
-  validates_presence_of :title, :body
+  validates_presence_of :title, :body, :criteria_rules
```

Comment:
If we are validating the presence of criteria_rules here we should probably remove `allow_nil: true` from line 9 below - `validates :criteria_rules, criteria_schema: true, allow_nil: true`?

Reply:
It's ok to still have the allow_nil here as that just means that the criteria_schema validation isn't run if the value is nil. 

Though it looks like the validation won't error if the input is nil so I'll remove it.

Reply:
Oh I see, I thought the `allow_nil: true` meant that nil was an allowed value to be saved (hence my comment). But thinking about it, nil is allowed anyway unless there is a db constraint so ignore me.